pkg/common/config/config.go

```diff
@@ -735,3 +735,31 @@ func GetConfigPath(ctx context.Context) string {
 	}
 	return cfgPath
 }
+
+// GetSessionUserAgent returns clusterwise unique useragent
+func GetSessionUserAgent(ctx context.Context, useragent string) string {
+	log := logger.GetLogger(ctx)
+	if strings.TrimSpace(useragent) == "" {
+		useragent = "k8s-csi-useragent"
+	}
+	clusterFlavor, err := GetClusterFlavor(ctx)
+	if err != nil {
+		log.Errorf("failed retrieving cluster flavor. Error: %v", err)
+		return useragent
+	}
+	cfg, err := GetConfig(ctx)
+	if err != nil {
+		log.Errorf("failed to read config. Error: %+v", err)
+		return useragent
+	}
+	if clusterFlavor == cnstypes.CnsClusterFlavorVanilla {
+		if cfg.Global.ClusterID != "" {
+			useragent = useragent + "-" + cfg.Global.ClusterID
+		}
+	} else {
+		if cfg.Global.SupervisorID != "" {
+			useragent = useragent + "-" + cfg.Global.SupervisorID
```

Comment:
make sure we read this when cluster flavor is for workload (supervisor).

Reply:
added check for  workload cluster flavor.